cheroot/test/test_ssl.py

```diff
@@ -625,6 +628,41 @@ def test_ssl_env(  # noqa: C901  # FIXME
     )
 
 
+class TestBufferedWriterSSLWantErrors:
+    """Test SSL want error handling in BufferedWriter."""
+
+    def setup_method(self):
+        """Set up test fixtures with mock raw socket and BufferedWriter."""
+        self.mock_raw = mock.Mock()
+        self.mock_raw.closed = False
+        self.writer = BufferedWriter(self.mock_raw)
+
+    def test_want_write_error_retry(self):
+        """Test that WantWriteError causes retry with same data."""
+        test_data = b'hello world'
+
+        self.mock_raw.write.side_effect = [
+            OpenSSL.SSL.WantWriteError(),
+            len(test_data),
+        ]
+
+        result = self.writer.write(test_data)
```

Comment:
Could you name the vars meaningfully? Stuff like $output, $result, $thing is too generic to make sense of.